test/unit/workflow_test.rb

```diff
@@ -57,7 +57,7 @@ class WorkflowTest < ActiveSupport::TestCase
   end
 
   test 'generates fresh RO crate for individual workflow' do
-    workflow = Factory(:cwl_workflow, license: 'MIT', other_creators: 'Jane Smith, John Smith')
+    workflow = Factory(:cwl_workflow, license: 'https://opensource.org/licenses/MIT', other_creators: 'Jane Smith, John Smith')
```

Comment:
This looks wrong. The licenses are stored as key which then then turned into a URL as part of the schema.org generation. They shouldn't be directly stored as a URL

Reply:
Does this mean we can change them to point to spdx?

Reply:
yes, we already could. https://github.com/seek4science/seek/blob/seek-1.11/lib/seek/bio_schema/resource_decorators/creative_work.rb#L23
or possibly use identifier.org

Reply:
if this is to get the url for the ro-crate, then it would be better to make the change there rather than changing the behaviour of Workflow to be different from other assets. You could add a licence_url to all licensed assets

Reply:
I'm not clear what you are suggesting.

Reply:
OK. 
You seem to have overridden the behaviour for Workflow and then manipulated the tests to store the full URL, but in reality it would still be the key. 
This seems to be so that the ro-crate generated would have a license URL by reading workflow.license but I think this will only work for the test, due to the full URL being directly stored.
You would need to call `Seek::License.find(workflow.license)&.url` for the ro-crate too, but rather than duplicating this for both ro-crate and the schema.org you could create a new `license_url ` method that does this lookup in one place.

Reply:
https://github.com/seek4science/seek/blob/workflow/lib/seek/workflow_extractors/ro_crate.rb#L61

would also need to be changed to convert an incoming license URL to a license ID that SEEK recognizes

Reply:
so is `workflow.license` currently being stored as either a URL or ID depending if it was submitted as an RO-Crate or not ? Entries I've looked at through the JSON API use the ID, e.g https://workflowhub.eu/workflows/52.json

Reply:
it's always an ID at the moment, but this is not valid according to bioschemas or schema.org

I mention it here: https://about.workflowhub.eu/Workflow-RO-Crate/#supported-licenses

Reply:
yes, in the bioschema.org markup generated for the page it was getting translated to a URL from the stored ID. e.g


```<script type="application/ld+json">{
"@context": "http://schema.org",
"@type": "ComputationalWorkflow",
"@id": "https://workflowhub.eu/workflows/52",
...
"license": "https://opensource.org/licenses/MIT",
...
```
it just needs to do the same for the RO-Crate to get the same URL and be valid

as long as the ID's at https://about.workflowhub.eu/Workflow-RO-Crate/#supported-licenses are the same as opendefinition and spdx which they seem to be, it should be ok.
It is possible to change the URL conversion to create SPDX or Indentifiers.org URLs for software licenses in the future quite easily.

@alaninmcr's change will mean the invalid ID will be shown in both the page markup and, from what I understand, also the RO Crate

Reply:
https://about.workflowhub.eu/Workflow-RO-Crate/#supported-licenses could be changed to list of recommended SPDX urls instead, and they could be easily translated back to ID's seek recognises.

Reply:
OK. The spdx suggest that the SPDX IDs should be used whereever possible. There is a JSON file that contains the list of spdx licenses https://github.com/librariesio/spdx/blob/master/licenses.json

I think we should leverage that file.,

Reply:
Just for this problem, I have changed to use the constant e.g. 'MIT'. I have not made the change to spdx